Student/Student/Submissions/SubmissionRubric/RubricCircleView.swift

```diff
@@ -33,76 +36,72 @@ class RubricCircleView: UIView {
         return formatter
     }()
 
-    static func computedHeight(rubric: RubricViewModel, maxWidth: CGFloat) -> CGFloat {
-        let count = CGFloat(rubric.ratings.count)
-        let howManyCanFitInWidth = CGFloat( floor( maxWidth / (w + space) ) )
-        let rows = CGFloat(ceil(count / howManyCanFitInWidth))
-        return (rows * w) + ((rows - 1) * space)
-    }
+    func setupButtons() {
+        //  remove old buttons
+        buttons.forEach { $0.removeFromSuperview() }
+        buttons = []
 
-    override func draw(_ rect: CGRect) {
         let w = RubricCircleView.w
         let space: CGFloat = 10
         let ratings: [Double] = rubric?.ratings ?? []
-        let howManyCanFitInWidth = Int( floor( rect.size.width / (w + space) ) )
+        let howManyCanFitInWidth = Int( floor( frame.size.width / (w + space) ) )
         let count = ratings.count
 
-        let ctx = UIGraphicsGetCurrentContext()
-        var center = CGPoint(x: w / 2, y: w / 2)
+        var center = CGPoint(x: 0, y: 0)
 
         for i in 0..<count {
             let r = ratings[i]
             let selected = i == (rubric?.selectedIndex ?? 0)
 
+            let button = DynamicButton(frame: CGRect(x: center.x, y: center.y, width: w, height: w))
+            button.tag = i
+            button.addTarget(self, action: #selector(actionButtonClicked(sender:)), for: .touchUpInside)
+            button.layer.cornerRadius = floor( w / 2 )
+            button.layer.masksToBounds = true
+                let title = RubricCircleView.formatter.string(for: r) ?? ""
```

Comment:
Is this just github doing weird indentation?

Reply:
it's off in code, fixed, thanks for pointing this out :hides: 😂